src/ServicePulse.Host/app/js/views/message/controller.js

```diff
@@ -0,0 +1,155 @@
+            vm.loadMessage(messageId).then(function () { vm.togglePanel(vm.message, 1); });
+            
+        };
+
+        notifier.subscribe($scope, function (event, messageFailureResolved) {
+            if (messageFailureResolved.failed_message_id === vm.message.id) {
+                toastService.showInfo('Message was successfully retried.');
+                vm.message.retried = false;
+                vm.message.resolved = true;
+            }
+        }, "MessageFailureResolvedByRetry");
+
+        notifier.subscribe($scope, function (event, messageFailureResolved) {
+            if (messageFailureResolved.failed_message_id === vm.message.id) {
+                toastService.showInfo('Message failed.');
+                vm.message.retried = false;
+            }
+        }, "MessageFailed");
+
+        vm.clipComplete = function(messageId) {
+            toastService.showInfo(messageId + ' copied to clipboard');
+        };
+
+        vm.togglePanel = function (message, panelnum) {
+            if (!message)
+                return false;
+
+            if (!angular.isDefined(message.messageBody)) {
+                serviceControlService.getMessageBody(message.message_id).then(function (msg) {
+                    message.messageBody = msg.data;
+                }, function () {
+                    message.bodyUnavailable = "message body unavailable";
+                });
+            }
+
+            if (!angular.isDefined(message.messageHeaders)) {
+                serviceControlService.getMessageHeaders(message.message_id).then(function (msg) {
+                    message.messageHeaders = msg.data[0].headers;
+                }, function () {
+                    message.headersUnavailable = "message headers unavailable";
+                });
+            }
+            message.panel = panelnum;
+            return false;
+        };
+
+        vm.retryMessage = function () {
+            serviceControlService.retryFailedMessages([vm.message.id])
+                .then(function() {
+                        toastService.showInfo("Retrying the message " + vm.message.message_id + " ...");
+                        vm.message.retried = true;
+                    }
+                );
+        };
+
+        vm.archiveMessage = function () {
+            serviceControlService.archiveFailedMessages([vm.message.id])
+                .then(function() {
+                    toastService.showInfo("Archiving the message " + vm.message.message_id + " ...");
+                    // below line is a way to not fetch for the whole message from SC. We update date to now and calculate delete fields
+                    vm.message.last_modified = $moment().format();
+                    calculateDeleteDate(vm.message, vm.error_retention_period); 
+                    vm.message.archived = true;
+                });
+        };
+
+        function calculateDeleteDate(message, errorRetentionPeriod) {
```

Comment:
`calculate` to me means a single result (an answer) will be provided, but this method mutates the incoming object.


Maybe the name should be `updateMessageDeleteDate` ?